google/resource_compute_instance.go

```diff
@@ -762,6 +770,15 @@ func resourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("can_ip_forward", instance.CanIpForward)
 	d.Set("machine_type", GetResourceNameFromSelfLink(instance.MachineType))
 
+	// PROVISIONING, STAGING, RUNNING, STOPPING, STOPPED,
+	//	// SUSPENDING, SUSPENDED, and TERMINATED
+	switch instance.Status {
```

Comment:
Instead of flattening several states into either `RUNNING` or `STOPPED`, let's see if we can approach this from a different direction.

Right now, we use an `Operation` from the GCP API to tell if the instance has been provisioned. That doesn't tell us the status of the logical instance, just of the GCP resource. What if we also waited in `Create` for the instance to be `RUNNING` if the field is set to `RUNNING`, and stop it if it's set to `TERMINATED`.

Then, in `Read` we can wait to enter a terminal value and d.Set the exact value.

Finally, in `Update` if we're in a different state than expected we need to start / stop depending on which terminal state we want to be in. There's some additional nuance around this section; https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance.go#L1253

We'll also need to ensure that `TERMINATED ` instances can be deleted.

The intent of this change would be that we're explicitly entering a terminal state in each of the CRUD methods, so Terraform will always be in a well-known place, and we won't need to worry about how we can transition from `PROVISIONING` -> `TERMINATED` for example.